Add papers citing or using Mitiq to the documentation

nathanshammah opened the pull request:
Description
-----------
Add a subsection in the documentation with a list of papers citing or using Mitiq. Include links to Semantic Scholar and Google Scholar to keep it up to date. Close #424. When the dependency graph will show something, we can add a link to https://github.com/unitaryfund/mitiq/network/dependents.

Checklist
-----------
Please make sure you have finished the following tasks before requesting a review of the pull request (PR). For more information, please refer to the [Contributor's Guide](../blob/master/CONTRIBUTING.md):

- [ ] Contributions to mitiq should follow the [pep8 style](https://www.python.org/dev/peps/pep-0008/). You can enforce it easily with [`black`](https://black.readthedocs.io/en/stable/index.html) style and with [`flake8`](http://flake8.pycqa.org) conventions.
- [ ] Please add tests to cover your changes, if applicable, and make sure that all new and existing tests pass locally.
- [x] If the behavior of the code has changed or new feature has been added, please also [update the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md).
- [ ] Functions and classes have useful [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings and [type hints](https://www.python.org/dev/peps/pep-0484/) in the signature of the objects.
- [ ] (Bug fix) The associated issue is referenced above using [auto-close keywords](https://help.github.com/en/articles/closing-issues-using-keywords). If the PR fixes an issue, use the keyword fix/fixes/fixed followed by the issue id. For example, if the PR fixes issue 1184, type "Fixes #1184" (without quotes).
- [x] The [changelog](https://github.com/unitaryfund/mitiq/blob/master/CHANGELOG.md) is updated, including author and PR number (@username, gh-xxx).

rmlarose reviewed (changes requested):
What about leaving out specific papers and only linking to google/semantic scholar? Otherwise we'd just have to update everything in 6 months.

Also, can we make this more prominent in the docs? The `citing` page is a bit buried, and it now has two purposes - how to cite Mitiq and how to find papers citing Mitiq. What about adding the google/semantic scholar links to `guide-overview`?

nathanshammah commented:
@rmlarose I went with a list just because right now the Semantic Scholar and GScholar citations are scattered and two papers are not picked up by neither one. I agree with you (https://github.com/unitaryfund/mitiq/issues/424#issuecomment-723002566) but my idea for the start was to keep the list updated, e.g., ~quarterly, and then discontinue it as you suggest. Also, the idea of having a couple of papers was nice as more self-contained, e.g., in the pdf. 

Anyways is fine for me, but I will proceed as decided by the majority based on the others' have opinions (@crazy4pi314 @andreamari) as the issue was already about adding the list and I want to make sure we all agree. 

I agree on the proposal to make this more evident in the docs.

andreamari commented:
Having an explicit list of papers at this preliminary stage sounds good to me.
When this becomes hard to manage (I am looking forward to this situation! ) we can switch to a link.

Marketing considerations  :smiley: :
A positive aspect of the explicit list is that it gives more visibility to papers citing Mitiq (a kind of free advertising for the authors).
Another positive aspect is that in this way the references get indexed in Google. So, if someone search for a paper citing Mitiq, our docs will appear (maybe).

rmlarose commented:
Ok, an explicit list is fine for me, but I unvolunteer myself from having to update it :)

So this LGTM once said list is more visible in the docs.

crazy4pi314 commented:
I am ok with explicit list until we can get a correct auto feed, is there a way to add the missing ones?

Also as to making it more prominent, would making the section called Mitiq Research in stead of Citing help?  We could also add it to the Readme under citing, and similarly rename?

andreamari reviewed (approved):
LGTM!

nathanshammah commented:
@andreamari @crazy4pi314 @rmlarose thanks all for the comments. Please review the updates. @rmlarose: I added a section to the readme, and renamed the "Citing"--> "Research" as proposed by @crazy4pi314, and more information in the Guide overview. I agree that there is some possible change in the future we could make with regards to the structure of what goes in the Users Guide and what not, but is better to branch it into another issue, eventually.

nathanshammah commented:
@crazy4pi314 regarding the updating the list, if the question is if there is a way to tell Google Scholar to add papers to its graph, no, there isn't. I had a quick look online for ways to embed citing works, but they mostly are framed for researchers (list of authored papers + citations), instead of list of other papers citing one's work, so decided to desist.

rmlarose commented on docs/source/guide/guide-overview.rst:
```diff
@@ -15,3 +15,8 @@ library with a long-term vision to be useful for quantum programmers using any q
 `Qiskit <https://qiskit.org/>`_, and `PyQuil <https://pyquil-docs.rigetti.com/en/stable/>`_
 frontends and backends.
+
+The guide contains information of how :ref:`zero noise extrapolation <guide_zne>` can be implemented with the software and examples of how it can be applied to any simulator or quantum processor running :ref:`Back-end Plug-ins <guide-executors>`.
```
My preference is to not include these because they are very apparent in the left/top nav and forward/back buttons. It's one less thing to have to update in the docs.

nathanshammah replied:
Fine. Here I just updated the guide overview to reflect the new sections, it contained zne and executors since I added the research link as you suggested, which looked a bit awkward without information on the rest of the guide sections.

rmlarose commented on docs/source/guide/guide-overview.rst:
```diff
@@ -15,3 +15,8 @@ library with a long-term vision to be useful for quantum programmers using any q
+
+The guide contains information of how :ref:`zero noise extrapolation <guide_zne>` can be implemented with the software and examples of how it can be applied to any simulator or quantum processor running :ref:`Back-end Plug-ins <guide-executors>`.
+
+A brief overview with more information about error mitigation techniques from the research literature can be found :ref:`here <guide_qem>` (instead, the :ref:`examples` section shows how Mitiq can be used in your :ref:`research`.)
+
```
Same here, preference to leave the TOC auto-generated.

rmlarose commented on docs/source/research.rst:
```diff
@@ -0,0 +1,72 @@
+Citing Mitiq
+------------
+
+If you are using ``Mitiq`` for your research, please cite the related `white paper <https://arxiv.org/abs/2009.04417>`_:
```
Remove ticks around Mitiq

rmlarose commented on docs/source/research.rst:
```diff
@@ -0,0 +1,72 @@
+You can download the :download:`bibtex file <mitiq.bib>`.
+
+
+If you have developed new features for error mitigation, or found bugs in ``mitiq``, please consider `contributing <contributing.html>`_ your code.
```
Use Mitiq instead of ``mitiq``

andreamari replied:
Maybe we can remove this line, since we already talk about new features and feature contributions above.

crazy4pi314 reviewed (approved):
LGTM! Thanks @nathanshammah

rmlarose commented on docs/source/research.rst:
```diff
@@ -0,0 +1,72 @@
+========
+Research
+========
+Mitiq is designed in order to aid researchers in quantum computing and quantum error mitigation. Given the quick growth of the techniques for quantum error mitigation (you can read an overview in the `Users Guide <https://mitiq.readthedocs.io/en/stable/guide/guide-error-mitigation.html>`_ ), it is natural to update the toolchain with new techniques and features. If you'd like to propose a new fea ...
```
Remove "in order"

rmlarose commented on docs/source/research.rst:
```diff
@@ -0,0 +1,72 @@
+-------------------------
+Contributing new features
+-------------------------
+If you have some code or have a plan to add it to Mitiq, discuss it in an issue and once ready to upload the code, review the `contributing <contributing.html>`_ guidelines for the steps to take.
```
"If you have some code"? Maybe "If you'd like to contribute new features to Mitiq, ..."?

rmlarose reviewed (commented):
We now have the Mitiq bib entry in three separate places - readme, research, and mitiq.bib. Isn't it better to, say, put it in a single file (e.g., the bib) and link to this file everywhere?

The new research page has a lot of redundancy with the contributing docs.

andreamari commented on README.md:
```diff
@@ -135,8 +139,12 @@ If you use Mitiq in your research, please reference the [Mitiq preprint][arxiv]
+
 [arxiv]: https://arxiv.org/abs/2009.04417
 
+[papers_with_mitiq]: https://mitiq.readthedocs.io/en/stable/research.html#papers-citing-or-using-mitiq
```
Now this link doesn't work but I guess the idea is that it will work after we do the next release, right?

nathanshammah replied:
yes.

andreamari reviewed (approved):
I left a minor comment and an optional suggestion. LGTM!

nathanshammah commented:
@rmlarose @andreamari I implemented suggested changes, thanks (besides the bibtex repetition, which looks not to bad to me).
